test/test_sync.cpp

```diff
@@ -6204,6 +6204,95 @@ TEST(Sync_DeleteCollectionInCollection)
     }
 }
 
+TEST(Sync_NestedCollectionClear)
```

Comment:
Let's add the same test for dictionaries too

Reply:
Done

Reply:
I was hoping we can have the same schema with different scenarios as for lists (i.e, Dictionary<Mixed> vs Dictionary in Mixed)

Reply:
I am not sure what you mean. I have added a test case that fails without the fix that this PR provides.

Reply:
I meant that for lists you use all three Mixed fields in the schema in tests.